core/modules/replica/WorkerSqlRequest.h

```diff
@@ -0,0 +1,181 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_WORKERSQLREQUEST_H
+#define LSST_QSERV_REPLICA_WORKERSQLREQUEST_H
+
+// System headers
+#include <string>
+
+// Qserv headers
+#include "replica/protocol.pb.h"
+#include "replica/WorkerRequest.h"
+
+// Forward declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+namespace database {
+namespace mysql {
+    class Connection;
+}}}}} // namespace lsst::qserv::replica::database::mysql;
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * Class WorkerSqlRequest executes queries against the worker database
+ * and return results sets (if any) back to a caller.
+ *
+ * @note
+ *   queries passed into this operation must be well formed.
```

Comment:
What happens if they are not?

Reply:
The query execution will fail and the database exception `replica::database::mysql::Error` with an explanation of a problem with the syntax of the query would be thrown. The will be caught by:
```
try {
    ...
} catch(database::mysql::Error const& ex) {
    ..
}````
within a method executing the query:
```
bool WorkerSqlRequest::execute() {
```
And be reported back to the calling `Controller` in `Protobuf` structure:
```
message ProtocolResponseSql {
    ...
    /// MySQL-specific error
    optional string error = 5 [default = ""];
    ...
}
```
I'm going to extend the `Doxygen` documentation for the class to make it clear.